Description: Inject libatomic dependency to fix armel build as in #918060
Author: Paolo Greppi <paolo.greppi@libpf.com>
Last-Update: 2019-09-15

Index: doxygen/src/CMakeLists.txt
===================================================================
--- doxygen.orig/src/CMakeLists.txt
+++ doxygen/src/CMakeLists.txt
@@ -309,6 +309,9 @@ if (use_libclang)
     llvm_map_components_to_libnames(llvm_libs core option)
     target_compile_definitions(doxygen PRIVATE ${LLVM_DEFINITIONS})
     set(CLANG_LIBS libclang clangTooling ${llvm_libs})
+    if ($ENV{DEB_HOST_ARCH} STREQUAL "armel")
+        set(EXTRA_CLANG_LIBS atomic)
+    endif()
 endif()
 
 target_link_libraries(doxygen
@@ -326,6 +329,7 @@ target_link_libraries(doxygen
     ${EXTRA_LIBS}
     ${CLANG_LIBS}
     ${DOXYGEN_EXTRA_LINK_OPTIONS}
+    ${EXTRA_CLANG_LIBS}
 )
 
 set_project_warnings(doxycfg)
Index: doxygen/addon/doxyparse/CMakeLists.txt
===================================================================
--- doxygen.orig/addon/doxyparse/CMakeLists.txt
+++ doxygen/addon/doxyparse/CMakeLists.txt
@@ -15,6 +15,9 @@ doxyparse.cpp
 
 if (use_libclang)
     set(CLANG_LIBS libclang clangTooling ${llvm_libs})
+    if ($ENV{DEB_HOST_ARCH} STREQUAL "armel")
+        set(EXTRA_CLANG_LIBS atomic)
+    endif()
 endif()
 
 target_link_libraries(doxyparse
@@ -31,6 +34,7 @@ ${CMAKE_THREAD_LIBS_INIT}
 ${SQLITE3_LIBRARIES}
 ${EXTRA_LIBS}
 ${CLANG_LIBS}
+${EXTRA_CLANG_LIBS}
 )
 
 install(TARGETS doxyparse DESTINATION bin)
